Watchman.AwsResources/Services/Sqs/QueueDataV2Source.cs

```diff
@@ -1,67 +1,74 @@
     {
         private readonly QueueSource _queueSource;
+        private const string ErrorQueueSuffix = "_error";
 
         public QueueDataV2Source(QueueSource queueSource)
         {
             _queueSource = queueSource;
         }
 
-        private Task<IList<string>> ReadActiveQueueNames()
-        {
-           return _queueSource.GetResourceNamesAsync();
-        }
-
         protected override string GetResourceName(QueueDataV2 resource)
         {
             return resource.Name;
         }
 
         protected override async Task<IEnumerable<QueueDataV2>> FetchResources()
         {
-            var names = await ReadActiveQueueNames();
+            var queueNames = await _queueSource.GetResourceNamesAsync();
 
-            var queues = names
-                .Where(e => !IsErrorQueue(e))
-                .Select(n =>
-                {
-                    var errorQueueName = names.FirstOrDefault(
-                        e => e.StartsWith(n) &&
-                             IsErrorQueue(e));
+            var uniqResourcesNames = queueNames
+                .Select(i => GetResourceName(i))
+                .Distinct();
+
+            var result = new List<QueueDataV2>();
+            foreach (var resourcesName in uniqResourcesNames)
+            {
+                var workingQueueName = resourcesName;
+                var errorQueueName = $"{resourcesName}{ErrorQueueSuffix}";
 
+                var hasWorkingQueue = queueNames.Any(i => string.Equals(workingQueueName, i));
+                var hasErrorQueue = queueNames.Any(i => string.Equals(errorQueueName, i));
 
-                    // we could have a scenario where nothing has been published/read to/from the error queue for
-                    // a few weeks so CloudWatch stops reporting it. In this case we still do want to alert on the queue
-                    // so in this case we can guess the name as it should be predictable anyway.
-                    errorQueueName = errorQueueName ?? $"{n}{ErrorQueueSuffix}";
+                var resource = new QueueDataV2(resourcesName);
 
-                    return new QueueDataV2()
-                    {
-                        Name = n,
+                if (hasWorkingQueue)
+                {
+                    resource.SetWorkingQueue(workingQueueName);
+                    resource.SetErrorQueue(errorQueueName);
+                }
+
+                if(!hasWorkingQueue && hasErrorQueue)
+                    resource.SetErrorQueue(errorQueueName);
 
-                        ErrorQueue = new QueueDataV2()
-                        {
-                            Name = errorQueueName
-                        }
-                    };
-                });
+                result.Add(resource);
+            }
 
-            return queues;
+            return result;
         }
 
-        private bool IsErrorQueue(string queueName)
+        private static string GetResourceName(string queueName)
```

Comment:
could we either inline this method or rename it? took me a while to figure out that it was stripping off the _Error suffix

Reply:
done